base/complex.jl

```diff
@@ -424,7 +424,8 @@ function log2(z::Complex)
     a/log(oftype(real(a),2))
 end
 
-function exp(z::Complex)
+exp(z::Complex) = exp(float(z))
```

Comment:
And also the (probably) new preferred way of adding similar test one line above the one you added.